etc/ci_scripts/dump_artifacts/nomad_artifacts.py

```diff
@@ -60,25 +61,37 @@ class NomadTask:
     parent: NomadAllocation = dataclasses.field(repr=False)
 
     def get_logs(self, nomad_client: Nomad, type: OutOrErr) -> str:
-        return cast(
-            str,
-            nomad_client.client.stream_logs.stream(
-                self.parent.allocation_id, self.name, type=type, plain=True
-            ),
-        )
+        try:
+            return cast(
+                str,
+                nomad_client.client.stream_logs.stream(
+                    self.parent.allocation_id, self.name, type=type, plain=True
+                ),
+            )
+        except URLNotFoundNomadException as e:
+            LOGGER.info(f"Couldn't get logs for {self.name}")
+            return None
+
+
+def _get_nomad_job_names(nomad_client: Nomad) -> List[str]:
+    jobs: List[str] = [j["ID"] for j in nomad_client.jobs if not j["ParameterizedJob"]]
```

Comment:
Is the job id really the job name in Nomad?
Also why are we excluding the parameterized jobs?

Reply:
A) That is correct
B) For parameterized jobs, you will see the following:
- integration_tests (param batch job)
- integration_tests/dispatch-12338838 (a dispatched instance of integration_tests)

We don't want to bother "getting logs" for the uninstanced `integration_tests` (but also, I'll document this)